### PR TITLE
Versioning Strategy

### LINKED ISSUE #68: API versioning and clear deprecation strategies
<p></p>

- Version strategy

- API changes

- Documents

<p></p>

<https://jamboard.google.com/d/15mzwIHN5t-g_e8BCs9fzWHQSrT6PV-_6X5xcR6edfvo/viewer?f=0>

### PR DESCRIPTION
**Pull request by kjgarza**
## Purpose

This is a new product specification. 
To review go to **Files changed** and click **Display the rich Diff**

closes: https://github.com/datacite/tasks/issues/68

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (feature that would cause existing functionality to change)

- [ ] New Product (non-breaking change which adds a new service)

## Reviewer, please remember to:

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: product_specifications/SDerr4/readme.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+Solution Validation: We run a feedback session with a number REST API users and gather their feedback.
+
+
+As up today that would be:
+
+Relese Candiate
+- https://api.datacite.org paths:
+  - /v3/dois
+  - /v3/repositories
+  - /v3/organisations
```
Comment (sarala) [inline product_specifications/SDerr4/readme.md:19]: Is organisation replacing providers? Are we deprecating /providers?

File: product_specifications/SDerr4/readme.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+For Deprecation
+- https://api.datacite.org, paths:
+  - /works
+  - /clients
+  - /users
```
Comment (sarala) [inline product_specifications/SDerr4/readme.md:25]: What about /providers?

Comment (sarala) [reply product_specifications/SDerr4/readme.md:25]: and /members

File: product_specifications/SDerr4/readme.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            When the user make a call the API endpoint 
+            Then response header should include the Custom header 'X-Datacite-API-Release-Candiate' with a url linking to the changelog of the announcing the release candiate
+
+
+Feature: Depraction of Endpoints in V3
+
+        Scenario: API user goes to /v3/clients or /v3/users or /v3/providers or /v3/members endpoint 
+            Given that a API call is made to /v3/clients or /v3/users or /v3/providers or /v3/members
+            When a API call is made to /v3/clients or /v3/users or /v3/providers or /v3/memberss
+            Then the API reponse be 404 should include the message "The resource you are looking for doesn't exist."
```
Comment (sarala) [inline product_specifications/SDerr4/readme.md:70]: I see here we are deprecating providers and members. You might want to add this to the list above.

File: product_specifications/SDerr4/support_doc.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- MDS
+- EZID
+
+## Release schedule
+
+Datacite releases a new API version only when is deemed strictly necessary.
+
+Each stable version is supported for a minimum of 12 months. When a new stable version is introduced and contains changes that affect your app, you have 18 months to test and migrate your app to the new version before support for the previous version is set for deprecation. All deprecation introduction would be announced in the DataCite Newsletter and/or DataCite Open Hours.
+We strongly recommend updating your apps to call the latest stable API version. 
+If your request doesn't include a version, then the API also defaults to the oldest supported stable version.
```
Comment (sarala) [inline product_specifications/SDerr4/support_doc.md]: May be 'oldest supported stable version' -> 'last stable version'

File: product_specifications/SDerr4/support_doc.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+REST DOIs API URLs: /{ version }/dois/{ id }
+
+For example, the following URLs call version 3:
+
+REST DOIs API URL: /v3/dois/10.2323/102
+REST DOIs (collection) API URL: /v3/dois
+
+
+There are several supported versions of the APIs available, and you specify the version that you want to use by substituting the version name in the URL. There are three types of API versions: `stable`, `release candidate`, and `unsupported`.
```
Comment (sarala) [inline product_specifications/SDerr4/support_doc.md]: Where does the 'deprecated' fit in?

Comment (kjgarza) [reply product_specifications/SDerr4/support_doc.md]: that would be `unsupported` maybe I can normalize the language.

File: product_specifications/SDerr4/readme.md (hunk 1 above)
Comment (kjgarza) [reply product_specifications/SDerr4/readme.md:19]: so version 3 won't have providers

**Pull request merged**